packages/mds-stream-processor/labelers/geography-labeler.ts

```diff
@@ -15,29 +15,59 @@
  */
 
 import db from '@mds-core/mds-db'
-import { UUID, Nullable, Telemetry } from '@mds-core/mds-types'
+import { UUID, Nullable, Telemetry, Geography } from '@mds-core/mds-types'
 import { pointInShape } from '@mds-core/mds-utils'
+import getBbox from '@turf/bbox'
+import { BBox } from '@turf/helpers'
 import { MessageLabeler } from './types'
 
 export interface GeographyLabel {
   geography_ids: UUID[]
 }
 
-export const GeographyLabeler: () => MessageLabeler<
-  { telemetry?: Nullable<Telemetry> },
-  GeographyLabel
-> = () => async ({ telemetry }) => {
-  const gps = telemetry?.gps
-  if (gps) {
-    const { lat, lng } = gps
+type WithBbox<T> = T & { bbox: BBox }
 
-    const geographies = await db.readGeographies({ get_published: true })
+export const pointInBbox = ({ lat, lng }: { lat: number; lng: number }, bbox: BBox) => {
+  const [bottomLeft, topRight] = [
+    [bbox[0], bbox[1]],
+    [bbox[2], bbox[3]]
+  ]
 
-    const geography_ids = geographies
-      .filter(({ geography_json }) => pointInShape({ lat, lng }, geography_json))
-      .map(({ geography_id }) => geography_id)
+  return bottomLeft[0] <= lng && bottomLeft[1] <= lat && topRight[0] >= lng && topRight[1] >= lat
+}
+
+const computeBbox = (geography: Geography) => {
+  return getBbox(geography.geography_json)
+}
+
+const newGeos = (cachedGeos: Map<UUID, boolean>, readGeos: Geography[]) =>
+  readGeos.filter(readGeo => !cachedGeos.has(readGeo.geography_id))
+
+export const GeographyLabeler: () => MessageLabeler<{ telemetry?: Nullable<Telemetry> }, GeographyLabel> = () => {
+  const cachedGeos: WithBbox<Geography>[] = []
+  const cachedGeosMap: Map<UUID, boolean> = new Map()
```

Comment:
I'm not a huge fan of maintaining both of these, but for the sake of runtime performance having the hash map of geographies we've already saved is pretty important.

Reply:
Did you consider using `MemCache` like the device labeler does? Would need to move it.

Reply:
I thought about it, but the usage is kind of different -- in the device labeler case it's easier because we're trying to reference a device from the in-memory cache, and fallback to redis cache then db. In this case, we need to be constantly reading the geographies from the db and then validate against what's already in the cache.